lib/consyncful/item_mapper.rb

```diff
@@ -25,17 +25,26 @@ def id
       @item.id
     end
 
-    def mapped_fields(locale)
+    def mapped_fields(default_locale)
       fields = generic_fields
 
       @item.fields_with_locales.each do |field, value_with_locales|
-        value = value_with_locales[locale.to_sym]
-        next if value.is_a? Contentful::File # it is special
+        value_with_locales.each do |locale_code, value|
+          next if value.is_a? Contentful::File # assets are handeled below
 
-        assign_field(fields, field, value)
+          fieldname = locale_code == default_locale.to_sym ? field : "#{field}_#{locale_code.to_s.underscore}".to_sym
+
+          assign_field(fields, fieldname, value)
+        end
       end
 
-      fields[:file] = raw_file(locale) if type == 'asset'
+      if type == 'asset'
+        raw_files_by_locale.each do |locale_code, details|
+          fieldname = locale_code.to_sym == default_locale.to_sym ? :file : "file_#{locale_code.to_s.underscore}".to_sym
+
+          fields[fieldname] = details
+        end
+      end
 
       fields
     end
```

Comment:
I think ideally this guy probably could have it's two logical parts split into private methods instead of adding to the rubocop todos.

Could be something like:

```rb
def mapped_fields(default_locale)
  fields = generic_fields
  fields.merge(localized_fields)
  fields.merge(localized_asset_fields) if type == 'asset'
  fields
end

private

def localized_fields
  @item.fields_with_locales.each do |field, value_with_locales|
    value_with_locales.each do |locale_code, value|
      next if value.is_a? Contentful::File # assets are handeled below

      fieldname = locale_code == default_locale.to_sym ? field : "#{field}_#{locale_code.to_s.underscore}".to_sym
      assign_field(fields, fieldname, value)
    end
  end
end

def localized_asset_fields
  raw_files_by_locale.each do |locale_code, details|
    fieldname = locale_code == default_locale ? 'file' : "file_#{locale_code.to_s.underscore}"
    fields[fieldname.to_sym] = details
  end
end
```

Reply:
This then opens up an opportunity to inline the `raw_files_by_locale` method (which you've simplified but could even go further with a `dig`? What're your thoughts? 😄 

```rb
def localized_asset_fields
  files_by_locale = @item.raw.dig('fields', 'file') || {}
  @item.raw.fetch raw_files_by_locale.each do |locale_code, details|
    ...
```

Reply:
Nice suggestion - done!